src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -98,23 +98,32 @@ citus_table_is_visible(PG_FUNCTION_ARGS)
 
 
 /*
- * RelationIsAKnownShard gets a relationId, check whether it's a shard of
- * any distributed table. If onlySearchPath is true, then it searches
+ * RelationIsAKnownShard gets a shardRelationOid, checks whether it's a shard
+ * of any distributed table. If onlySearchPath is true, then it only searches
  * the current search path.
- *
- * We can only do that in MX since both the metadata and tables are only
- * present there.
  */
 bool
-RelationIsAKnownShard(Oid shardRelationId, bool onlySearchPath)
+RelationIsAKnownShard(Oid shardRelationOid, bool onlySearchPath)
 {
-	bool missingOk = true;
-	char relKind = '\0';
+	Oid ownerRelationOid = GetOwnerRelationOid(shardRelationOid, onlySearchPath);
+
+	return OidIsValid(ownerRelationOid);
+}
 
-	if (!OidIsValid(shardRelationId))
+
+/*
+ * GetOwnerRelationOid gets a shardRelationOid, returns OID of the relation
+ * that owns the shard relation with shardRelationOid if it's a valid shard
+ * of it. If onlySearchPath is true, then it only searches the current search
```

Comment:
what is meant by current search path?

Reply:
just searching through the current `search_path`, means the set of schemas that is set by doing `set search_path to ...`